### PR TITLE
test pyright ouput lines vs rdjson output

### PR DESCRIPTION
**Pull request by Chadsr**
I have been seeing some issues with the reviewdog annotation being displayed on the line below where it should be, so I started writing a test suite for the rdjson converter function.

So far the test is passing as expected, so this is a WIP whilst I figure out where the line mismatch might be coming from. (Either way, the script gets a test suite, even if the problem is in reviewdog/pyright itself!)

### CODE REVIEW

**Comment by Chadsr** [comment]
Closes #15

**Comment by jordemort** [comment]
@Chadsr Is this one ready to go? If so, could you hit the "Ready for review" button? Thanks!

**Comment by Chadsr** [comment]
@jordemort I was hoping to look further for this potential line offset issue, but i'm not sure if I have the time right now. So yeah, let's merge this and any fixes (if needed) can come along later.

**Comment by jordemort** [comment]
@Chadsr Upon review it looks like the linters are cranky about a few things; could you take a look at those annotations and appease it, or do you mind if I do it? I really should update the workflow so it does suggestions instead of just check failures. If you click over to the "Files changed" view you can see the things it is complaining about.

**Comment by Chadsr** [comment]
@jordemort Shoud be fixed now!

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # action-pyright
-
+[![test](https://github.com/jordemort/action-pyright/actions/workflows/test.yml/badge.svg)](https://github.com/jordemort/action-pyright/actions/workflows/test.yml)
```
Comment (Chadsr) [inline README.md:2]: I honestly have no idea what mardownlint-fix thinks is wrong here?

Comment (jordemort) [reply README.md:2]: I think it wants a blank line between the heading and the content

**Comment by jordemort** [comment]
Sorry, dropped off the planet for a bit but I'm back

File: README.md (hunk 1 above)
Comment (Chadsr) [reply README.md:3]: Indeed it was

**Pull request merged**